go/libkb/util.go

```diff
@@ -865,3 +865,27 @@ func NoiseXOR(secret [32]byte, noise NoiseBytes) ([]byte, error) {
 func ForceWallClock(t time.Time) time.Time {
 	return t.Round(0)
 }
+
+// Decode decodes src into dst.
+// Errors unless all of:
+// - src is valid hex
+// - src decodes into exactly len(dst) bytes
+func DecodeHexFixed(dst, src []byte) error {
+	// hex.Decode is wrapped because it does not error on short reads and panics on long reads.
+	if len(src)%2 == 1 {
+		return hex.ErrLength
+	}
+	if len(dst) != hex.DecodedLen(len(src)) {
+		return NewHexWrongLengthError(fmt.Sprintf(
+			"error decoding fixed-length hex: expected %v bytes but got %v", len(dst), hex.DecodedLen(len(src))))
+	}
+	n, err := hex.Decode(dst, src)
+	if err != nil {
+		return err
+	}
+	if n != len(dst) {
```

Comment:
This should be impossible if `err` is `nil`, right? I'm not against checking it though.

Reply:
https://play.golang.org/p/XyCXAzazfBd

Reply:
My bad!